opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -261,6 +280,65 @@ pub trait ResourceDetector {
     fn detect(&self) -> Resource;
 }
 
+/// Builder for [Resource]
+#[derive(Debug)]
+pub struct ResourceBuilder {
+    resource: Resource,
+}
+
+impl ResourceBuilder {
+    /// Add a single [ResourceDetector] to your resource.
+    pub fn with_detector(self, detector: Box<dyn ResourceDetector>) -> Self {
+        self.with_detectors(vec![detector])
+    }
+
+    /// Add multiple [ResourceDetector] to your resource.
+    pub fn with_detectors(mut self, detectors: Vec<Box<dyn ResourceDetector>>) -> Self {
```

Comment:
Could we change the argument type from `Vec` to a slice? We don't have to force users to allocate a vec when they can create an array on stack.

Reply:
updated